src/libp2p/Libp2p.Core/IWriter.cs

```diff
@@ -9,7 +9,7 @@ namespace Nethermind.Libp2p.Core;
 
 public interface IWriter
 {
-    ValueTask WriteLineAsync(string str, bool prependedWithSize = true)
+    ValueTask<IOResult> WriteLineAsync(string str, bool prependedWithSize = true)
```

Comment:
What's the reason for `ValueTask` here?

Reply:
It's instantly completed most of the times, which is more efficient to handle with ValueTask. There are mid term plans to run and share benchmarks.